### ISSUE TITLE
iOS: erratic elements sizing with GridItemsLayout

### ISSUE THREAD

**Issue by SergTomcat** [dotnet/maui#13481]
### Description

When I use CollectionView with GridItemsLayout, it doesnt size properly (if images are present). Scrolling help fix layout.

![Simulator Screen Recording - iPhone 14 Pro - 2023-02-21 at 06 03 17](https://user-images.githubusercontent.com/24654961/220370515-d033aaf4-5358-43f5-a9d4-ccb135196850.gif)

Setting ItemSizingStrategy doesn't help.

Happen only on iOS.

Need fix or workaround (how to implement tile view or call collectionview rearrangement)

### Steps to Reproduce

1. Create Xaml with CollectionView:

```
	<CollectionView x:Name="ItemsCollectionView" ItemsSource="{Binding Items}" VerticalScrollBarVisibility="Always" HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand" ItemSizingStrategy="MeasureAllItems">
		<CollectionView.ItemTemplate>
			<DataTemplate>
				<VerticalStackLayout VerticalOptions="FillAndExpand" x:Name="ItemContainer">
					<Frame HeightRequest="{Binding Source={RelativeSource Self}, Path=Width}" HasShadow="False">
						<Image Source="{Binding Image, Mode=OneWay}" HorizontalOptions="FillAndExpand"  Aspect="AspectFit"/>
					</Frame>
					<Label Text="{Binding Name}" LineBreakMode="WordWrap" HorizontalOptions="CenterAndExpand"/>
				</VerticalStackLayout>
			</DataTemplate>
		</CollectionView.ItemTemplate>
		<CollectionView.ItemsLayout>
			<GridItemsLayout Orientation="Vertical" Span="2"/>
		</CollectionView.ItemsLayout>
	</CollectionView>
	
```
	
2. Provide data with ImageSource
	

---------------------------------
In reproduction repository run project, in side menu select "Tile view" entry.

	

### Link to public reproduction project repository

https://github.com/SergTomcat/MAUI_Bug_Demo

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16

### Did you find any workaround?

No workarounds.
Layout changes if after delay you set ItemSizingStrategy to MeasureAllItems, then after another delay MeasureFirstItem. But not always and not always correctly.

Scrolling sets items in correct order, but only while actual scrolling happen. Calling Scroll method without scrolling available doesn't help.

**Comment by drasticactions** [comment]
Your repo is targeting net6.0 and not 7.0. I upgraded it and the good news is that it does fix part of your layout issues...

<img width="359" alt="スクリーンショット 2023-02-22 0 14 29" src="https://user-images.githubusercontent.com/898335/220384308-a037baa1-9be2-42b7-ba02-d5392032a59b.png">

It does now lay out with the proper sizes, but it's now overlapping each other. Same happens on Catalyst. And since it's not within the scroll zone, you can't scroll to fix it, you need to rotate your phone to force it to refresh.

**Comment by SergTomcat** [comment]
> Your repo is targeting net6.0 and not 7.0. I upgraded it and the good news is that it does fix part of your layout issues...
>  
> It does now lay out with the proper sizes, but it's now overlapping each other. Same happens on Catalyst. And since it's not within the scroll zone, you can't scroll to fix it, you need to rotate your phone to force it to refresh.

My bad. I didn't upgrade this to net7, but my actual project is on net7.0 and there is exactly these overlapping tiles. If you double the amount of tiles (so scroll could appear) and scroll, they will rearrange perfectly. Haven't found the way to force it from code and without scroll though....

**Comment by drasticactions** [comment]
Looking into this...

https://github.com/SergTomcat/MAUI_Bug_Demo/blob/main/Xaminals/Views/TilePage.xaml#L10-L16

There are a few interesting things you're doing here. 

[`FillAndExpand` has no effect anymore](https://learn.microsoft.com/en-us/dotnet/maui/migration/layouts?view=net-maui-7.0) so I would remove it. The items inside the StackLayout need to be set with a size on their own, which does not help with trying to generate the layout. That's coming from the frame, where you set a `HeightRequest="{Binding Source={RelativeSource Self}, Path=Width}"` on the frame itself. This is, most likely, where the bug is in MAUI (It's most likely not setting this value properly by the time the layout is rendered, and then resizing or scrolling forces it to rerender, now with the right values). This would also apply to setting it with a `MaxHeight` value too.

If you want to work around this, you could change your StackLayout to a Grid and set a `HeightRequested` on it.

```
			<DataTemplate>
				<Grid HeightRequest="500" x:Name="ItemContainer">
					<Grid.RowDefinitions>
						<RowDefinition Height="*"/>
						<RowDefinition Height="Auto"/>
					</Grid.RowDefinitions>
					<Frame Grid.Row="0" HasShadow="False">
						<Image Source="{Binding Image, Mode=OneWay}" Aspect="AspectFit"/>
					</Frame>
					<Label Grid.Row="1" Text="{Binding Name}" LineBreakMode="WordWrap" HorizontalOptions="CenterAndExpand"/>
				</Grid>
			</DataTemplate>
```

That'll force it to load with a height and fix the initial layout.

**Comment by SergTomcat** [comment]
@drasticactions yes, fixed height arranges them right. Problem is I have requirement to make frame perfectly square and square edge is dependant on screen width.
If there is no workaround to make CollectionView "catch" size dynamically or force it to redraw, then I will have to precalculate screen width and set it to Frame height (for iOS, on Android it works fine), which is sad, because it moves all style spacings  (paddings, margins) from styles to actual code...

**Comment by jinxinjuan** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 1.0. Can repro on iOS platform with sample project.
https://github.com/SergTomcat/MAUI_Bug_Demo
![image](https://github.com/dotnet/maui/assets/37974803/75a74ba2-a6f5-4b5a-97f0-7e40c2761764)

**Comment by HarishwaranVijayakumar** [comment]
As the repo link is not working. I have created a sample and tested. This issue is not reproducible in the latest .NET MAUI version 9.0.90 on iOS 18.0 in both CV1 and CV2. I have attached video for your reference. 

[Repo.zip](https://github.com/user-attachments/files/21749131/Repo.zip)

https://github.com/user-attachments/assets/a6b52198-14a1-46f1-9d60-80e1abbe78ef
